sync2/atxs_test.go

```diff
@@ -0,0 +1,393 @@
+package sync2_test
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/jonboulle/clockwork"
+	"github.com/stretchr/testify/require"
+	gomock "go.uber.org/mock/gomock"
+	"go.uber.org/zap/zaptest"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/spacemeshos/go-spacemesh/common/types"
+	"github.com/spacemeshos/go-spacemesh/fetch"
+	"github.com/spacemeshos/go-spacemesh/p2p"
+	"github.com/spacemeshos/go-spacemesh/p2p/pubsub"
+	"github.com/spacemeshos/go-spacemesh/sync2"
+	"github.com/spacemeshos/go-spacemesh/sync2/rangesync"
+	"github.com/spacemeshos/go-spacemesh/sync2/rangesync/mocks"
+	"github.com/spacemeshos/go-spacemesh/system"
+)
+
+func atxSeqResult(atxs []types.ATXID) rangesync.SeqResult {
+	return rangesync.SeqResult{
+		Seq: func(yield func(k rangesync.KeyBytes) bool) {
+			// Received sequence may be cyclic and the handler should stop
+			// when it sees the first key again.
+			for _, atx := range atxs {
+				if !yield(atx.Bytes()) {
+					return
+				}
+			}
```

Comment:
The idea is that it can be cyclic from consistency standpoint, `rangesync.Seq` being _possibly_ cyclic.
As described in other comment, I may try to get rid of this, but that takes some effort

Reply:
Got rid of cyclicity considerations here